src/Repository/Repository.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Repository;
+
+use App\Controllers\Home\HomeController;
+use App\Repository\DBConnexion;
+use PDO;
+
+abstract class Repository
+{
+    protected DBConnexion $dbConnection;
+    protected $table;
+    protected $entity;
+
+    public function __construct(DBConnexion $dbConnection, $table, $entity)
+    {
+        $this->dbConnection = $dbConnection;
+        $this->table = $table;
+        $this->entity = $entity;
```

Comment:
can you rename entity to entityClass , when I read entity I think it's an object